### PR TITLE
Add WebSocket Echo server example

### PR DESCRIPTION
**Pull request by ChrisGuzman**

### CODE REVIEW

File: voice/echo_server.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+get '/ncco' do
+  content_type :json
+  [
+    {
+      "action": "connect",
+      "eventUrl": [
+        "https://example.com/events"
+      ],
+      "from": "441632960960",
```
Comment (leggetter) [inline voice/echo_server.rb]: Should be `FROM_NUMBER` pulled in via an environment variable.

File: voice/echo_server.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  end
+end
+
+get '/ncco' do
+  content_type :json
+  [
+    {
+      "action": "connect",
+      "eventUrl": [
+        "https://example.com/events"
```
Comment (leggetter) [inline voice/echo_server.rb]: Can we determine the correct URL for this server and use it?

File: voice/echo_server.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      ],
+      "from": "441632960960",
+      "endpoint": [
+        {
+          "type": "websocket",
+          "uri": "ws://example.com/socket",
```
Comment (leggetter) [inline voice/echo_server.rb]: Can we determine the correct URL for this server and use it?

File: voice/echo_server.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
+require 'faye/websocket'
+require 'sinatra'
+require 'thin'
+require 'json'
+
+# To run this code sample you need to install: gem install faye-websocket sinatra thin
```
Comment (leggetter) [inline voice/echo_server.rb]: Add to Gemfile?

**Comment by ChrisGuzman** [comment]
@leggetter this is ready for re-review

File: voice/websocket_echo_server.rb [hunk 5]
```diff
@@ -46,13 +50,13 @@
     {
       "action": "connect",
       "eventUrl": [
-        "https://example.com/events"
+        "#{base_url}/events"
```
Comment (leggetter) [inline voice/websocket_echo_server.rb]: 👍

File: voice/websocket_echo_server.rb [hunk 6]
```diff
@@ -46,13 +50,13 @@
...
       ],
-      "from": "441632960960",
+      "from": "#{FROM_NUMBER}",
```
Comment (leggetter) [inline voice/websocket_echo_server.rb]: 👍

File: voice/websocket_echo_server.rb [hunk 7]
```diff
@@ -46,13 +50,13 @@
...
       "endpoint": [
         {
           "type": "websocket",
-          "uri": "ws://example.com/socket",
+          "uri": "ws://#{request.env['HTTP_HOST']}/socket",
```
Comment (leggetter) [inline voice/websocket_echo_server.rb]: 👍

**Pull request merged**